contracts/staking/GoodFundManager.sol

```diff
@@ -156,7 +169,42 @@ contract GoodFundManager is DAOContract {
         Reward memory reward = Reward(_rewardsPerBlock, _blockStart, _blockEnd, _isBlackListed);
         rewardsForStakingContract[_stakingAddress] = reward;
     }
-   
+    /**
+     * @dev Add active contract to active contracts array
+     * @param _stakingContract address of the staking contract
+     */
+    function addActiveStakingContract(address _stakingContract) public {
+        _onlyAvatar();
+        //check if address exists in array
+        bool exist;
+        for (uint8 i=0; i < activeContracts.length; i++){
+            if(activeContracts[i] == _stakingContract){
+                exist = true;
+                break;
+            }
+        }
+        require(exist == false , "Staking contract address already exist");
+        activeContracts.push(_stakingContract);
+    }
+    /**
+     * @dev Remove active contract from active contracts array
+     * @param _stakingContract address of the staking contract
+     */
+    function removeActiveStakingContract(address _stakingContract) public {
+        _onlyAvatar();
+        uint index;
+        bool exist;
+        for (uint8 i=0; i < activeContracts.length; i++){ 
+            if(activeContracts[i] == _stakingContract){
+                exist = true;
+                index = i;
+                break;
+            }
+        }
+        require(exist==true, "There is no such a address to delete");
+        activeContracts[index] = activeContracts[activeContracts.length - 1];
+        activeContracts.pop();
+    }
```

Comment:
we dont need this. it should be part of setStakingReward. if reward>0 && !blacklisted then add to active, otherwise remove.
when you loop over active contracts in setStakingReward you can also remove those that blockend>now

Reply:
active contracts should be array in order use in collectinterest so we can't use mapping since we cant iterate

Reply:
i ment that we dont need this functions. the logic should be in setStakingReward

Reply:
Ok will change them

Reply:
Also maybe I should check if there is interest in the staking contract or not before I remove it? Because this logic will conflict with this one https://github.com/GoodDollar/GoodProtocol/pull/59#discussion_r626372597 @sirpy